cmd/kube-scheduler/app/server.go

```diff
@@ -616,7 +619,7 @@ func (s *SchedulerServer) Run(stop chan struct{}) error {
 			return fmt.Errorf("couldn't create leader elector: %v", err)
 		}
 
-		leaderElector.Run()
+		leaderElector.Run(context.TODO())
```

Comment:
Why is the controller manager Background and scheduler TODO contexts?

Reply:
Because currently it blocks forever and I didn't want to change any behaviour in this pr. I guess the right thing to do would be to cancel the context when the `stop` is closed/done. To do that we'd need to start a goroutine that blocks on `stop` and cancels the context when unblocked. Will do that.